### ISSUE TITLE
Problem on cdaql1 with podd/SConstruct when CODA env defined 

### ISSUE THREAD

**Issue by MarkKJones** [JeffersonLab/analyzer#144]
When running scons on the cdaql1 computer, I had the problem that in podd
it could not find the evio library. This is because in podd/SConstruct it checks to see if CODA env variable
is defined.  If CODA is defined then it assumes that the evio library and include files are there.
On cdaql1, it is defined as /site/coda/2.6.2 but there are no evio library or header file in
Linux/lib and Linux/include. 

It seems that we need to add a check that the  evio libraries and header files can be found rather than just check that the CODA variable is defined. I CODA evio not found then use the evio
gotten from github.  I do not think we can assume the same CODA directory structure everywhere.

**Comment by hansenjo** [comment]
I just went with what the DAQ group does. When compiling EVIO, it
installs itself in $CODA by default (unless you set --prefix explicitly).

Ole

**Comment by hansenjo** [comment]
How about having the build system look in $EVIO first, then in $CODA?

**Comment by hansenjo** [comment]
And yes, if evio.h is not present in $EVIO/include or $CODA/include, then we download the source and build it locally.

**Comment by brash99** [comment]
Sure, this can be done.  I’ll work on it ... probably Monday before I can get to it in earnest.

Cheers,
E.

Dr. Edward J. Brash
Department of Physics, Computer Science & Engineering
Christopher Newport University
work: (757) 594-7451
cell: (757) 753-2831
www.cnu.edu/pcs

**Comment by hansenjo** [comment]
I added support for detecting an actual EVIO installation under $EVIO_LIBDIR/$EVIO_INCDIR, $EVIO and $CODA to the Make system in commit 924b792. See the commit message for details. I'll let Ed handle the SCons part.

**Comment by MarkKJones** [comment]
Until it gets fixed the quick fix to the SCONS problem is
edit  the podd/SConstruct

Change line 

coda_dir = os.getenv('CODA')

to

coda_dir = os.getenv('CODAtemp')

This will cause it to download evio instead of looking in CODA.

**Comment by brash99** [comment]
I should be able to get to this one this afternoon.

**Comment by pooser** [comment]
For folks that are working on the cdaq cluster and using the non standard ROOT PRO installation (5.34.36) you will run into issues regarding a clean installation of hcana.  From what I understand you will also run into issues utilizing the PRO installation however that is aside the point.

Currently, when one logs into cdaql1 the default ROOT is source to the PRO version.  If one then switches to say 6.08.00 via. the command ‘source /apps/root/6.08.00/setroot_CUE.csh’ then your LD_LIBRARY_PATH and PATH environment variables will still contain stale links to the PRO version of ROOT.  With the newest updates to Scons, this will cause the build of hcana to fail.

The hack to solve this issue is that one needs to remove the PRO version of ROOT in the aforementioned paths.  This can be accomplished with the following shell commands:

setenv PATH `echo $PATH | perl -pe “s[/apps/root/PRO/root/bin[]g;”`
setenv LD_LIBRARY_PATH `echo $LD_LIBRARY_PATH | perl -pe “s[/apps/root/PRO/root/lib[]g;”` 

Once this is done, perform a clean scons build and everything should work…famous last words :-p

**Comment by pooser** [comment]
There should be a ` character in front of echo.  I could not figure out how to disable the code insertion character in the thread.

**Comment by brash99** [comment]
OK, I think I have fixed SConstruct to check this properly now (first checking to see if EVIO_LIBDIR/EVIO_INCDIR are defined, then if EVIO is defined, then if CODA is defined).  In each case, it checks to see if the specific evio.h and libevio.* files are present.  Here is the commit:  

I just added these changes to the previous pull request that I already had open.  Or at least I hope that I did.

Cheers,
E.
